packages/haul-ram-bundle-webpack-plugin/src/__tests__/integration.test.ts

```diff
@@ -213,30 +247,154 @@ describe('should build and successfully evaluate bundle', () => {
   });
 });
 
-async function enhanceFrameWithSourceMaps(frame: string, indexSourceMap: any) {
-  const [
-    ,
-    name,
-    file,
-    line,
-    column,
-  ] = /^at (.+) \((\d\.js):(\d+):(\d+)\)$/.exec(frame)!;
-  const moduleSourceMap = indexSourceMap.sections.find(
-    (section: any) => section.map.file === file
-  );
-  assert(moduleSourceMap, `Couldn't find source map for file ${file}`);
-  const consumer = await new SourceMapConsumer(moduleSourceMap.map);
-  const {
-    name: origName,
-    column: origColumn,
-    line: origLine,
-    source,
-  } = consumer.originalPositionFor({
-    column: parseInt(column, 10),
-    line: parseInt(line, 10),
-    bias: SourceMapConsumer.LEAST_UPPER_BOUND,
+describe('File RAM bundle should build and successfully evaluate bundle', () => {
+  beforeAll(() => {
+    jest.setTimeout(10000);
   });
-  return `at ${origName || name} (${source}:${origLine}:${
-    origColumn ? origColumn + 1 : column
-  })`;
-}
+
+  it('with namedModules:false', async () => {
+    await build();
+
+    const bootstrap = fs.readFileSync(BUNDLE_PATH).toString();
+    expect(bootstrap).toBeDefined();
+    const script = new vm.Script(bootstrap, { filename: BUNDLE_PATH });
+
+    let context: vm.Context;
+    const global = {
+      nativeRequire: (localId: number, segmentId: number) => {
+        // pack module id from local id and segment id
+        const moduleId = (segmentId << 16) + localId;
+        const source = fs
+          .readFileSync(
+            path.join(path.dirname(BUNDLE_PATH), `js-modules/${moduleId}.js`)
+          )
+          .toString();
+        vm.runInContext(source, context);
+      },
+      global: undefined,
+      done: undefined,
+      crash: undefined,
+    };
+    (global as any).global = global;
+    context = vm.createContext(global);
+
+    let error: Error | undefined;
+    try {
+      script.runInContext(context);
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error).toBeUndefined();
+    expect(global.done).toBe(true);
+
+    await wait(1);
+
+    expect(global.crash).toBeDefined();
+    expect(() => {
+      ((global as any) as { crash: Function }).crash();
+    }).toThrow('test error');
+  });
+
+  it('with namedModules:true', async () => {
```

Comment:
there's quite some duplicated code here. consider using `test.each` for less repetition